Library/Homebrew/utils/curl.rb

```diff
@@ -238,6 +238,48 @@ def curl_check_http_content(url, specs: {}, user_agents: [:default], check_conte
       "The URL #{url} may be able to use HTTPS rather than HTTP. Please verify it in a browser."
     end
 
+    # Separates cURL response text into headers and body content. Headers (if
+    # any) are split into an array, where each member is a string of headers
+    # for an individual response.
+    # @param response_text [String] The response text from `curl` containing
+    #   headers, body, or both
+    # @return [Array<Array<String>, String, nil>] An array containing an array
+    #   of headers and the body content.
+    def response_headers_and_body(response_text)
+      headers = []
+      while response_text.match?(%r{\AHTTP/.* (\d+)})
+        response_text = response_text.lstrip
+        header_text, _, response_text = response_text.partition("\r\n\r\n")
+        headers << header_text.chomp if header_text.present?
+      end
+
+      [headers, response_text.lstrip]
+    end
+
+    # Extracts the status code from the final response along with the URL of
+    # the last `location` header encountered (if any).
+    # @param headers [Array<String>, String]
+    # @param url [String, nil] The URL to use as a base for making the
+    #  `location` URL absolute
+    # @param absolutize [true, false] Whether to make the location URL absolute
+    # @return [Array<String, nil>]
+    def response_status_code_and_location(headers, url: nil, absolutize: false)
+      # Separate the headers into individual responses if we're given a string
+      headers, = response_headers_and_body(headers) if headers.is_a?(String)
+
+      # Identify the last status code and location
+      status_code, final_location = nil
+      headers.each do |response_headers|
```

Comment:
Can `final_location` be set multiple times? If not, perhaps `.find` or `.select` or similar would make more sense here.

Reply:
What may not be clear here is that `headers` is an array of strings, where each string is the headers for a given response. This is explained in the comment before `response_headers_and_body`.

If there are multiple redirections, you will end up with more than one response with a `location` header. In this scenario, `final_location` is set on each iteration where a `location` header is present and the last value is what's returned as `final_location`.

Alternatively, if we only want to set `status_code` and `final_location` once (and `break` once this is satisfied), we can simply iterate in reverse:

```ruby
status_code, final_location = nil
headers.reverse_each do |response_headers|
  status_code ||= response_headers[%r{HTTP/.* (\d+)}, 1]
  location = response_headers[/^Location:\s*(.*)$/i, 1]&.chomp
  next unless location

  final_location = absolutize ? URI.join(url, location).to_s : location
  break
end
```

This would do less work but it's maybe a little less straightforward. If you think it's fine, I'm happy to replace it.